lib/mongodb.js

```diff
@@ -483,6 +483,25 @@ MongoDB.prototype.updateOrCreate = function updateOrCreate(model, data, options,
 };
 
 /**
+ * Replace if the model instance exists with the same id or create a new instance
+ *
+ * @param {String} model The model name
+ * @param {Object} data The model instance data
+ * @param {Object} options The options object
+ * @param {Function} [cb] The callback function
+ */
+MongoDB.prototype.replaceOrCreate = function(model, data, options, cb) {
```

Comment:
Hey Simon, I applied all of your feedback except changing `model` to `modelName` in many places you mentioned this change makes the code inconsistent with the rest of the code; in other words, in other methods others are using `model` as well and refactoring the other code written by others seems huge changes...